packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -61,109 +128,74 @@ interface TooltipProps {
-   * (default: 300)
-   */
-  skipDelayDuration?: number;
   children?: React.ReactNode;
 }
 
 const Tooltip: React.FC<TooltipProps> = (props: ScopedProps<TooltipProps>) => {
   const {
-    __scopeTooltip,
+    __scopeTooltipProvider,
     children,
     open: openProp,
     defaultOpen = false,
     onOpenChange,
-    delayDuration = 700,
-    skipDelayDuration = 300,
+    delayDuration: delayDurationProp,
   } = props;
-  const popperScope = usePopperScope(__scopeTooltip);
+  const context = useTooltipProviderContext(TOOLTIP_NAME, __scopeTooltipProvider);
+  const popperScope = usePopperScope(__scopeTooltipProvider);
   const [trigger, setTrigger] = React.useState<HTMLButtonElement | null>(null);
   const contentId = useId();
+  const openTimerRef = React.useRef(0);
+  const delayDuration = delayDurationProp ?? context.delayDuration;
+  const openDelay = context.isOpenDelayed ? delayDuration : 0;
+  const wasOpenDelayed = usePrevious(Boolean(openDelay));
+  const { onTooltipOpen, onTooltipClose } = context;
   const [open = false, setOpen] = useControllableState({
     prop: openProp,
     defaultProp: defaultOpen,
     onChange: onOpenChange,
   });
-  const [stateAttribute, setStateAttribute] = React.useState<StateAttribute>(
-    openProp ? 'instant-open' : 'closed'
-  );
 
-  // control open state using state machine subscription
-  React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
-      if (state === 'open' && context.id === contentId) {
-        setOpen(true);
-      } else {
-        setOpen(false);
-      }
-    });
-
-    return unsubscribe;
-  }, [contentId, setOpen]);
-
-  // sync state attribute with using state machine subscription
-  React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
-      if (context.id === contentId) {
-        if (state === 'open') {
-          setStateAttribute(context.delayed ? 'delayed-open' : 'instant-open');
-        } else {
-          setStateAttribute('closed');
-        }
-      } else {
-        setStateAttribute('closed');
-      }
-    });
-
-    return unsubscribe;
-  }, [contentId]);
-
-  const handleFocus = React.useCallback(
-    () => stateMachine.send({ type: 'FOCUS', id: contentId }),
-    [contentId]
-  );
-  const handleOpen = React.useCallback(
-    () => stateMachine.send({ type: 'OPEN', id: contentId, delayDuration }),
-    [contentId, delayDuration]
-  );
-  const handleClose = useCallbackRef(() => {
-    stateMachine.send({ type: 'CLOSE', id: contentId, skipDelayDuration });
-  });
+  const stateAttribute = React.useMemo(() => {
+    return open ? (wasOpenDelayed ? 'delayed-open' : 'instant-open') : 'closed';
+  }, [wasOpenDelayed, open]);
 
-  // send transition if the component unmounts
-  React.useEffect(() => () => handleClose(), [handleClose]);
+  const handleOpen = React.useCallback(() => {
+    document.dispatchEvent(new CustomEvent(TOOLTIP_OPEN));
```

Comment:
I dispatch here instead of within the `TooltipProvider` `onTooltipOpen` handler so that tooltips will still close if another tooltip is opened when there isn't a `TooltipProvider`.

This  actually made me wonder if we need a provider at all 🤔  I thought maybe I could use document events to control open delay but a first pass at it here isn't working. I'll take another look tomorrow but I need **one** `skipDelayTimerRef` across tooltips that they all can clear I think.

Reply:
I've had another go at this this morning, and I might be able to get it working but the code is a bit tricky to reason about. It also conceptually makes more sense in the provider I think, given that it's a shared delay between multiple tooltips.